product-service/src/handlers/getProductById.ts

```diff
@@ -0,0 +1,33 @@
+import { APIGatewayProxyHandler } from "aws-lambda";
+import productList from "../assets/productList.json";
+
+export const getProductById: APIGatewayProxyHandler = async (event) => {
+  try {
+    console.log("Lambda getProductById invocation with event: ", event);
+    const { productId } = event.pathParameters;
+    const product = productList.find((product) => product.id === productId);
+
+    if (product) {
+      return {
+        statusCode: 200,
+        headers: {
+          "Access-Control-Allow-Origin": "*",
+        },
+        body: JSON.stringify(product),
+      };
+    }
+
+    return {
+      statusCode: 404,
+      body: "Product not found",
+    };
+  } catch (error) {
+    return {
+      statusCode: 500,
+      headers: {
+        "Access-Control-Allow-Origin": "*",
+      },
+      body: "Something went wrong.",
+    };
+  }
+};
```

Comment:
acceptable to create one common error handler wrapper or decorator to catch data or errors instead of duplicate try-catch block in lambdas